tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -567,13 +572,23 @@ final class ScalaJSClassEmitter(semantics: Semantics,
 
   // Helpers
 
+  private def globalVarDef(name: js.Tree, value: js.Tree)(
+      implicit pos: Position): js.Tree = {
+    outputMode match {
+      case OutputMode.ECMAScript51Global =>
+        js.Assign(name, value)
+      case OutputMode.ECMAScript51Isolated =>
+        js.VarDef(name.asInstanceOf[js.VarRef].ident, value)
+    }
+  }
+
```

Comment:
Could we move this code, so it is physically closer to `envField`? I see that it is overkill to introduce a path dependent type on the output mode to not cast here. However, if the two methods were closer together, we reduce the risk of changing one without the other (also, a comment why the cast is OK, will be easily understandable).

Thinking of it, it might be better, if `globalVarDef` calls `envField` directly. The fact that only a tree from `envField` can be used is not even documented right now...